src/views/UIOptionsPopup/UIOptionsPopup.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import { useGlobalState } from '../../modules/global'
+import { UIOptionsDialog } from './style'
+import ValueSlider from '../Controls/ValueSlider'
+import ValueCheckbox from '../Controls/ValueCheckbox'
+import { Row, Col } from 'antd'
+import { CloseSquareOutlined } from '@ant-design/icons'
+
+type Props = {
+  children?: React.ReactNode
+}
+
+const UIOptionsPopup: React.FC<Props> = ({ children }: Props) => {
+  const [isVisible, setVisibility] = useGlobalState('isUIOptionsPopupVisible')
+
+  const showPopup = () => setVisibility(true)
+  const hidePopup = () => setVisibility(false)
+
+  const optionsDialog = isVisible ? (
+    <UIOptionsDialog>
```

Comment:
I like the tenacity to create a custom dialog popup. However for a project like this where we already have UI library installed, it's generally better to just use a component already available to you rather than making one from scratch. It both reduces code overhead and adds to UI uniformity on the app. I'd recommend using the antd modal for this one. Check out the docs here [https://ant.design/components/modal/](https://ant.design/components/modal/)

Reply:
TBH - I was having trouble getting the modal to work ... but I've learned a bit, so it'll probably go easier when I adapt to all of your suggestions :)

Reply:
I have submitted changes to honor your suggestions!